Formula/r/redress.rb

```diff
@@ -0,0 +1,63 @@
+class Redress < Formula
+  desc "Tool for analyzing stripped Go binaries compiled with the Go compiler"
+  homepage "https://go-re.tk/redress/"
+  url "https://github.com/goretk/redress/archive/refs/tags/v1.1.1.tar.gz"
+  sha256 "5c938791f2010233054ede11401ed7a00e32151f37042f95e08d103e46d98f53"
+  license "AGPL-3.0-only"
+  head "https://github.com/goretk/redress.git", branch: "develop"
+
+  livecheck do
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
+  depends_on "go" => [:build, :test]
+
+  def install
+    # https://github.com/goretk/redress/blob/develop/Makefile#L11-L14
+    redress_version = version
+    gore_version = File.read(buildpath/"go.mod").scan(%r{goretk/gore v(\S+)}).flatten.first
+    go_version = Formula["go"].version
+
+    ldflags = %W[
+      -s -w
+      -X main.redressVersion=#{redress_version}
+      -X main.goreVersion=#{gore_version}
+      -X main.compilerVersion=#{go_version}
+    ]
```

Comment:
Can't it figure out the go version from the compiler?
And what is the gore version for?
```suggestion
    # https://github.com/goretk/redress/blob/develop/Makefile#L11-L14
    gore_version = File.read(buildpath/"go.mod").scan(%r{goretk/gore v(\S+)}).flatten.first

    ldflags = %W[
      -s -w
      -X main.redressVersion=#{version}
      -X main.goreVersion=#{gore_version}
      -X main.compilerVersion=#{Formula["go"].version}
    ]
```

Reply:
These are flags taken directly out of the Makefile for the project; and match the implementation there.

I haven't deeply evaluated the source code to see if/how/where they make use of them.

Gore is a library it depends on.

Presumably it uses these for the version command or similar.

Reply:
Can we check what happens if we don't pass them?

Reply:
Anything is possible. Though I would ask why and whether it's important to spend time doing so.

Wouldn't it make the most sense to keep the way homebrew builds the program as close to the way it officially builds it?

That way it prevents introducing issues in the homebrew version through diverging from the official build.

Reply:
But if we wanted to do that, why not run the makefile? We're now adding a second implementation of the same logic that is up to the Homebrew community to keep in sync and maintain.

Reply:
Because if you look at the Makefile, it's just a whole lot of boilerplate to implement exactly what I've done here, but for 3 platforms.

I considered that Makefile; and decided it didn't make sense as it would complicate things for no gain. Using the Makefile here would be akin to saying to use a CI/CD workflow in the homebrew formula; it doesn't make sense.

Reply:
I'm confused.. I don't believe I said anything about not being open to reducing code here.

All I've done is answer your questions and provide relevant context?

Reply:
See this comment for latest updates, which sort of sidesteps the core testing issue being discussed here:

- https://github.com/Homebrew/homebrew-core/pull/155724#issuecomment-1839844939